### PR TITLE
Add GetPortByProcessId to allow parallel tests.

### LINKED ISSUE #2712: Add GetPortByProcessId to allow parallel tests

### PR DESCRIPTION
**Pull request by karthiknadig**
Fixes #2712

### CODE REVIEW

**Comment by karthiknadig** [comment]
@AlexanderSher Let me know if this needs any tweaks.

File: src/UnitTests/Core/Impl/ProcessUtils.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+                    err = GetTcpTable2(pBuf, ref nBufSize, false);
+                    if (err == 0) {
+                        int entrySize = Marshal.SizeOf(typeof(MIB_TCPROW2));
+                        int nEntries = Marshal.ReadInt32(pBuf);
+                        int tableStartAddr = (int)pBuf + sizeof(int);
+                        for (int i = 0; i < nEntries; i++) {
+                            IntPtr pEntry = (IntPtr)(tableStartAddr + i * entrySize);
+                            MIB_TCPROW2 tcpData = (MIB_TCPROW2)Marshal.PtrToStructure(pEntry, typeof(MIB_TCPROW2));
+                            if (tcpData.dwOwningPid == pid) {
+                                ports.Add(tcpData.dwLocalPort);
```
Comment (AlexanderSher) [inline src/UnitTests/Core/Impl/ProcessUtils.cs]: What are the `dwRemotePort` and `dwRemoteAddr` in this case?

File: src/UnitTests/Core/Impl/ProcessUtils.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+using System.Runtime.InteropServices;
+using static Microsoft.UnitTests.Core.NativeMethods;
+
+namespace Microsoft.UnitTests.Core {
+    public static class ProcessUtils {
+        public static List<int> GetPortByProcessId(int pid) {
+            List<int> ports = new List<int>();
+            int nBufSize = 0;
+            int err = GetTcpTable2(IntPtr.Zero, ref nBufSize, false);
+            if (err == 0x7a) { // ERROR_INSUFFICIENT_BUFFER
```
Comment (AlexanderSher) [inline src/UnitTests/Core/Impl/ProcessUtils.cs]: Can `err` be `0`?

Comment (karthiknadig) [inline src/UnitTests/Core/Impl/ProcessUtils.cs]: No. At minimum buffer has to be it has to be sizeof(int) + sizeof(int*), so no.

**Comment by AlexanderSher** [comment]
👍

File: src/UnitTests/Core/Impl/ProcessUtils.cs (hunk 1 above)
Comment (karthiknadig) [inline src/UnitTests/Core/Impl/ProcessUtils.cs]: suppose you start broker at http://localhost:5000 then
```
dwLocalAddr = 127.0.0.1 (in integer form)
dwLocalPort = 5000 (in integer form)
dwRemoteAddr = 127.0.0.1 (in integer form)
dwRemortPort = 0 (in integer form)
```

**Pull request merged**